lib/pure/asyncnet.nim

```diff
@@ -824,88 +822,58 @@ proc sendTo*(socket: AsyncSocket, data: pointer, dataSize: int, address: string,
 
   if not success:
     if lastException != nil:
-      retFuture.fail(lastException)
-
+      raise lastException
     else:
-      retFuture.fail(newException(
-            IOError, "Couldn't resolve address: " & address))
-
-  else:
-    retFuture.complete()
-
-proc sendTo*(socket: AsyncSocket, data, address: string, port: Port):
-            owned(Future[void]) {.async, since: (1, 3).} =
-  ## This proc sends ``data`` to the specified ``address``, which may be an IP
-  ## address or a hostname. If a hostname is specified this function will try
-  ## each IP of that hostname. The returned future will complete once all data
-  ## has been sent.
+      raise newException(IOError, "Couldn't resolve address: " & address)
+
+proc recvFrom*(socket: AsyncSocket, size: int,
+               flags = {SocketFlag.SafeDisconn}):
+              owned(Future[tuple[data: string, address: string, port: Port]])
+              {.async, since: (1, 3, 1).} =
```

Comment:
This is fine, but if you're up for it, a `recvFrom` taking a `FutureVar` and matching the net.recvFrom proc (https://nim-lang.org/docs/net.html#recvFrom%2CSocket%2Cstring%2Cint%2Cstring%2CPort%2Cint32) would be nice. Might be better to create it in a separate PR.

Reply:
Let me see if I understand, because my English is not the best.

Create another proc recvFrom (using net.recvFrom as a model), but using FutureVars in the parameters? In that, would be two recvFrom in asyncnet?

Reply:
yep, that's correct. You can do this in a separate PR.